### ISSUE TITLE
Music Conversion Logic

### ISSUE THREAD

**Issue by caseyjohnsonwv** [caseyjohnsonwv/playlist-pigeon#5]
The converter Lambda needs the functionality to actually convert songs between Spotify and Apple Music. Incoming requests will contain the below payload - this is not currently being validated by API Gateway (#6). 
```
{
  "source_service" : "spotify",
  "target_service" : "apple",
  "playlist_name" : "my spotify playlist",
  "track_list" : [
    "track-1-uuid",
    "track-2-uuid",
    "track-3-uuid",
  ]
}
```
Approximately. I'm sure this will need tweaking. Ideally the converter Lambda would work without credentials, only using public search APIs. The whole point of the asynchronous architecture is to stay within API rate limits from Spotify and Apple.

MVP:
1. Look up each track on the source service for its name and artist
2. Look up each track on the target service for its new UUID
3. Create a JSON file mapping UUIDs to UUIDs, source to target (eg, "track-1-uuid-spotify" : "track-1-uuid-apple")
4. Use the already implemented logic to upload the file to S3

**Comment by caseyjohnsonwv** [comment]
Using this space as a sounding board for my own design decisions:

It appears as though it would be much easier to simply receive a playlist URL in the backend, then lookup the playlist information instead of requiring this JSON payload. One problem - private playlists exist, and they require OAuth. Users will provide authorization in the frontend, meaning private playlists will be accessible there, but the backend (ideally) will run with only client credentials and public data.

TLDR - there's a reason I'm not just looking up the playlist in the backend, even though it seems like an obvious improvement.
